### PR TITLE
Ignore verbose if result-files-only

### LINKED ISSUE #23: If passing verbose and result-files-only, colcon test-result still shows the test results
I expect `colcon test-result --result-files-only --verbose` to show only the results file, but it lists the result files and the errors as well.

This limits the usefulness of `--result-files-only` when you have `verbose: true` in your defaults.yaml.

```
colcon test-result --result-file --verbose            
build/nav2_map_server/Testing/20190730-0331/Test.xml
build/nav2_map_server/test_results/nav2_map_server/test_occ_grid.gtest.xml
- nav2_map_server.MapLoaderTest loadInvalidFile
  <<< failure message
    /home/snuc/ros2_ws/src/navigation2/nav2_map_server/test/unit/test_occ_grid.cpp:218
    Expected: map_loader_->loadMapFromFile(loadParameters) throws an exception of type std::runtime_error.
      Actual: it throws a different type.
  >>>
build/nav2_map_server/test_results/nav2_map_server/uncrustify.xunit.xml
- nav2_map_server.uncrustify src/occ_grid_loader.cpp
  <<< failure message
    Diff with 26 lines
  >>>
```

### PR DESCRIPTION
**Pull request by rotu**
Also refactored some logic to make it clearer what the program does if result-files-only.

Fix #23

### CODE REVIEW

**Comment by dirk-thomas** [comment]
> Also refactored some logic ...

The changes in the logic seem to have broken some of the behavior, e.g. `--all`. Please keep refactorings separate from functional changes to simplify the review of each change and allow each to be merged independently.

**Comment by rotu** [comment]
Good call

File: colcon_test_result/verb/test_result.py [hunk 1]
```diff
@@ -56,23 +56,22 @@ def main(self, *, context):  # noqa: D102
...
-            print()
 
         summary = Result('Summary')
         for result in results:
             summary.add_result(result)
 
         if not context.args.result_files_only:
+            if (
+                    any(r.error_count or r.failure_count for r in results) or
+                    (context.args.all and results)
```
Comment (dirk-thomas) [inline colcon_test_result/verb/test_result.py]: Nitpick: these two lines should be indented one level (4 spaces from the `if`).

Comment (rotu) [reply colcon_test_result/verb/test_result.py]: ~~Weird. My IDE (PyCharm) did not like that indentation.~~ User error. I put it back the way you like it.

Comment (dirk-thomas) [reply colcon_test_result/verb/test_result.py]: It is less my "whim" but more like what PEP 8 defines...

**Comment by dirk-thomas** [comment]
Thanks for the patch!

**Pull request merged**